### PR TITLE
feat: Add performance tests for endpoints used in the WebUI initial load [WEB-1459]

### PR DESCRIPTION
**Pull request by julian-determined-ai**
## Description

Adds performance tests for the endpoints used in the initial load of WebUI pages. The API endpoints can be referenced in [this document](https://hpe.sharepoint.com/:w:/r/teams/detai/_layouts/15/Doc.aspx?sourcedoc=%7BAE5AA580-F855-4EC7-9268-0A9749DF43F2%7D&file=Identify%20API%20in%20Each%20Web%20Page.docx&action=default&mobileredirect=true).

Some tests will require references to existing data. For example existing experiments, trials, metrics, etc for `GET` requests. The tests are written with the expectation that those references will be passed into the testing script as environment variables. However, the test suite has been written so that all of the tests that can successfully run, will run, and all other tests will be skipped. 

### Example Test Skipping Output

Certain tests will be skipped since `RBAC` is not enabled, and no task id is specified. 
034BFC8D-0D30-438B-8D6D-A7F2607532A3_4_5005_c

### Example Console Output
91307070-AA18-4F5D-BCEE-A60A21AB804C

### Example jUnit output
72C2C941-DB60-4DCF-8596-474D80A5378E
28C16BCE-F1B3-4463-90FF-71FF1A3F7040_4_5005_c

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
WEB-1459

### CODE REVIEW

**Review by loksonarius — APPROVED** [review]

File: performance/src/utils/types.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+interface SeededData {
+    model: Model;
+    task: Task;
+    trial: Trial;
+    experiment: Experiment;
+    workspace: Workspace;
+}
+
+interface Workspace {
+    id?: string;
```
Comment (hkang1) [inline performance/src/utils/types.ts:46]: **question:** should this be required?

Comment (hkang1) [reply performance/src/utils/types.ts:46]: same question for Task.id and Trial.id definitions

File: performance/src/api_performance_tests.ts [hunk 2]
```diff
@@ -1,87 +1,544 @@
...
-        () => {
-            const res = http.get(generateEndpointUrl('/api/v1/master', clusterURL));
-            check(res, { '200 response': (r) => r.status == 200 });
-        }
-    ),
-]
+    const testSuite = [
+        // Query the master endpoint
+        test("get master configuration", () =>
+            testGetRequest("/api/v1/master", CLUSTER_URL, testConfig)),
```
Comment (hkang1) [inline performance/src/api_performance_tests.ts:71]: **nit suggestion:** one alternative to having to pass `CLUSTER_URL` and `testConfig` repetitively is to have a generative function approach:
```
export const testGetRequestor = (clusterUrl: string, testConfig?: TestConfiguration) => {
  return (url: string) => {
    const params = {
        headers: {
            "Content-Type": "application/json",
            Authorization: `${testConfig?.auth.token}`,
        },
    };
    const res = http.get(generateEndpointUrl(url, clusterURL), params);
    check(res, { "200 response": (r) => r.status == 200 });
  };
}
```
then you can just call:
```
const getRequest = testGetRequestor(CLUSTER_URL, testConfig);
...
getRequest("/api/v1/master");
...
```

**Review by hkang1 — APPROVED** [review]
LGTM! minor stuff

**Review by ashtonG — APPROVED** [review]

**Pull request merged**